text/0000-build-std/1-background.md

```diff
@@ -0,0 +1,309 @@
+immediately delegates to the panic handler crate.
+
+`std` is also a panic handler. `std`'s panic handler and `std::panic!` macro
+print panic information to stderr and delegate to a *panic runtime* to decide
+what to do next, determined by the *panic strategy*.
+
+There are two panic runtime crates in the standard library - `panic_unwind` and
+`panic_abort` - each with a corresponding panic strategy. Each target supported
+by rustc specifies a default panic strategy - either "unwind" or "abort" -
+though these are only relevant if `std`'s panic handler is used (i.e. the target
+isn't a `no_std` target or being used with a `no_std` crate).
+
+Rust's `-Cpanic` flag allows the user to choose the panic strategy, with the
+target's default as a fallback. If `-Cpanic=unwind` is provided then this
+doesn't guarantee that the unwind strategy is used, as the target may not
+support it.
+
+Both crates are compiled and shipped with the pre-built standard library for
+targets which support `std`. Some targets have a pre-built standard library with
+only the `core` and `alloc` crates, such as the `x86_64-unknown-none` target.
+While `x86_64-unknown-none` defaults to the `abort` panic strategy, as this
+target does not support the standard library, this default isn't actually
+relevant.
+
+The `std` crate has a `panic_unwind` feature that enables an optional dependency
+on the `panic_unwind` crate.
+
+`core` also has a `panic_immediate_abort` feature which modifies the
+`core::panic!` macro to immediately call the abort intrinsic without calling the
+panic handler. `std` and `alloc` have the same feature which enable the feature
+in `core`. `std`'s feature also adds an immediate abort to its `panic!` macro.
+
+## Cargo
+[background-cargo]: #cargo
+
+Cargo's building of the dependency graph is largely driven by the registry
+index, except for crates from `git` or `path` sources.
+
+[Cargo registries][cargo-docs-registry], like crates.io, are centralised sources
+for crates. A registry's index is the interface between Cargo and the registry
+that Cargo queries to know which crates are available, what their dependencies
+are, etc. crates.io's registry index is a Git repository -
+[rust-lang/crates.io-index] - which is updated automatically by crates.io when
+crates are published, yanked, etc. Cargo can query registries using a Git
+protocol which caches the registry on disk, or using a sparse protocol which
+exposes the index over HTTP and allows Cargo to avoid Cargo having a local copy
+of the whole index, which has become quite large for crates.io.
+
+Each crate in the registry has a JSON file, following
+[a defined schema][cargo-json-schema]. Crates may refer to those in other
+registries, but all non-`path`/`git` crates in the dependency graph must exist
+in a registry. As the registry index drives the building of Cargo's dependency
+graph, all non-`path`/`git` crates that end up in the dependency graph must be
+present a registry.
```

Comment:
Based on the intent given for this section, we should call out that
- the schema is shared between T-cargo and T-crates.io
- When a package is published, cargo posts a json blob that is not quite an index entry but is sufficient for generating one
- crates.io does not use the posted json blob but re-generates it from the `Cargo.toml` to avoid the Index and `Cargo.toml` from going out of sync from bugs or malicious publishes.  As a side effect, any Index change made in `cargo` has to be duplicated in crates.io 
  - On publish, crates.io actually extracts data from `Cargo.toml` and writes it to a database.  The Index and the frontend are then generated from that
- The frontend also renders dependency information